internal/yml/compare.go

```diff
@@ -38,3 +38,51 @@ func deepEqualNode(node1, node2 *yaml.Node) bool {
 
 	return true
 }
+
+func EqualWithExclusion(a, b *yaml.Node, excludedPath ...string) bool {
+	if a == nil && b == nil {
+		return true
+	}
+	if a == nil || b == nil {
+		return false
+	}
+
+	// Handling document nodes
+	if a.Kind == yaml.DocumentNode && b.Kind == yaml.DocumentNode {
+		if !EqualWithExclusion(a.Content[0], b.Content[0], excludedPath...) {
+			return false
+		}
+		return true
+	}
+
+	if a.Kind != b.Kind || a.Value != b.Value {
+		return false
+	}
+	if len(a.Content) != len(b.Content) {
+		return false
+	}
+
+	// Determine current key and rest of path
+	excludedKey := ""
```

Comment:
There's a couple issues here!
- I am finding it hard to follow this logic
- I am not able to prove to myself that its correct
- (Not important because its not our current need but...) This approach doesn't support sequences

On line 72 we remove a level from the excluded path, and pass it to line 82 regardless of if the key on the current level matches the removed index of the excluded path.

Therefore I think this actually ignores mapped items have the same amount of nesting as the given path and that have the same key as the last segment of the given path.

Also line 76 isn't compatible with nodes that have the empty string as their key even though that's an edge case we won't see, we can still consider it for correctness. 

Given that it is already hard to follow for my small brain, I wouldn't try and patch it with more if statements... Rather I would suggest you have a private implementation that recursively checks the logic and compares the excluded path against a current path.

```go
func EqualWithExclusion(a, b *yaml.Node, excludedPath []string) bool {
  return equalWithExclusion(a, b, excludedPath, nil)
}

func equalWithExclusion(a, b *yaml.Node, excluded, current []string) bool { ... }
```